src/main/resources/youtube/settings/host/values/AlbumCards.xml

```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```

Comment:
I understand your concerns regarding too many files and the maintenance aspect of it.

There's a structure of the "settings" resource directory hosting any strings for patches. It doesn't sound like a patch has to depend on `SettingsPatch` to merge some strings. An example would be introducing a patch that shows a toast but does not add in app settings and thus does not require the `Settings` patch.

My main motivation for separating the files is that each patch will be separated from each other, preserving the modular structure of patches. As of now, the translation patch hosts all the translations. While maintaining thousands of files would be problematic, these files would be generated automatically and never supposed to touch. Each patch would have a folder that would host the translations. Here, the folder structure is unnecessary; Crowdin could commit the translated files directly into the respective patches' translation folder, whereas the translation patch would pick them up and merge them to be added to the app. This would prevent having to create hundreds of folders, and instead, each patch would have its own translations under `<patchname>/translations`. Additionally, this structure preserves the modular structure of patches. If I want to remove a patch, for example, I can eliminate the resource directory for the patch and do not have to worry about first committing them, letting Crowdin register the removed strings and remove them from the translations.

The workflow would look the following way:

1. Create the resource directory for the translations (`<patchname>/translations>`)
2. Create the English `strings-en_US.xml` file
3. Commit and let Crowdin translate the files and commit the translations to `<patchname>/translations>`
4. The patches depend on the translations patch, which would pick up the translated files and create the `values` directories accordingly to write the strings file.

Let me know what you think.

Reply:
I believe it should be possible for Crowding to write translations to the same path of a source file. This will work for any number of source files. @Ushie can you confirm?

Reply:
Can you TL;DR? to the same directory? yes

Reply:
Basically, if I have multiple source strings files, each in separate directories, can Crowdin create translations for each of the source files in each of their directories?

Reply:
Yes, each imported source can have it's export directory managed individually

Reply:
@LisoUseInAIKyrios If there isn't anything else, we can start orchestrating anything that needs manual refactoring that I can help with.

Reply:
> It's a lot of clutter when each patch patch has 20 language files, but only one file can be edited

Yes. Each patch has its own language files. I primarily took that into account to preserve the modularity of patches. Because the files are automatically generated, it is fine that there are many language files. Maybe Crowdin has a feature that allows you to write all translations into a single file (per patch). This way, if we have 100 translatable patches, we will have 100 additional translated string files. The translation patch could then parse these files and merge all 100 translation files into one by mapping them to the corresponding language. But the prerequisite is that Crowdin supports that. If not, an intermediate branch Crowdin pushes to can trigger CI to apply the transformations. Otherwise, dealing with all the translation files is fine too, because they are generated automatically.

> The translation patch is also more clunky, because it has to merge 1,000 files and it needs to figure out which patches were included and which were not. 

It would have to iterate over each patch that it depends on, get all translated files, and map their translated strings to their respective language.

> if the root contains only a single subdirectory

You can disable flattening the path:

![image](https://github.com/ReVanced/revanced-patches/assets/13122796/28bb1ec8-2009-4518-afcd-2e976d88b9ea)

PS: You can use IntelliJ IDEA for all ReVanced projects, including integrations by importing it as a Gradle module, this gets rid of the need to use Android Studio.

Reply:
Please let me know if you still have any concerns.

Reply:
Each patch would have a resource directory attributed and named after itself, meaning, that you would be able to locate the string resource by querying the directory names with similar effort. Considering though, that the strings would be saved in the translations patch resource folder, I agree that it makes sense to skip this overhead and name the string files after the patch. I suggest to prefix or suffix it with "string" for example: "HideAds_strings.xml", "strings_HideAds.xml".

Regarding your previous response I still need some time to reply to it.